app/models/consultation.rb

```diff
@@ -196,7 +196,14 @@ def end_date_from(now = Time.zone.today)
     # Letters are printed at 5:30pm and dispatched the next working day (Monday to Friday)
     # Second class letters are delivered 2 days after they’re dispatched.
     # Royal Mail delivers from Monday to Saturday, excluding bank holidays.
-    default_start_date(now) + [consultee_response_period.days, period_days].max
+
+    effective_period_days = [consultee_response_period.days, period_days].max
+
+    if planning_application.application_type.include_bank_holidays?
+      Bops::Holidays.days_after_plus_holidays(from_date: default_start_date(now), count: effective_period_days)
+    else
+      default_start_date(now) + effective_period_days
+    end
```

Comment:
Just to confirm, when we say include bank holidays we are referring to treating these days as a normal day in the calculation right?
So it should behave like default. 
 
But when bank holidays are excluded then we factor these specific days in and the end date might be end up being a date more into the future than the default.

Not sure if I'm asking this properly so happy to have a chat 😄

Reply:
It's a very weird calculation. The normal situation is that the consultation is N days (default 21). That doesn't account for weekends or anything, it's just 21 calendar days.

In _some_ cases (I don't think we're sure which), then it's 21 calendar days _not counting bank holidays_. i.e,. if there's a single bank holiday within the 21 day period, the consultation actually ends 22 days later. (Weekends are still ignored for this calculation, so we can't say '21 business days' or anything like that.)

'Include bank holidays' is not necessarily a very obvious name for it but it feels hard to summarise the requirements.

Reply:
Or are you saying the meaning of the setting should be reversed: default (true) to be the normal behaviour of 21 days means 21 days, false means do the weird calculation?

That might make it more obvious, I'd just left it the way round it was when I'd added my own setting for this.

Reply:
Just on first pass, when I read include bank holidays I would assume it doesn't matter if it is a bank holiday, treat it like a normal day. But one could have the opposite interpretation too 🤔 
Agree that the naming is quite tricky with this.

Might be helpful if we can add some content for the user so we show specific guidance that if they check this checkbox then we will be calculating the consultation end date ignoring bank holidays or factoring them in (perhaps with an example). 
Then as long as we are consistent with what that flag is doing in our calculations it shouldn't matter much.

Reply:
This sounds like an outcome we can have from user testing

Reply:
Yeah, I did wonder if additional hint text might be useful for this option.